pkg/apis/calico/validation/networkconfig_test.go

```diff
@@ -0,0 +1,129 @@
+		Entry("should succeed with valid IPv6 pool mode", &apiscalico.NetworkConfig{IPv6: &apiscalico.IPv6{
+			Mode: ptr.To(apiscalico.Always),
+			Pool: ptr.To(apiscalico.PoolIPIP),
+		}}, field.NewPath("config"),
+			BeEmpty()),
+		Entry("should return error with invalid vethMTU", &apiscalico.NetworkConfig{VethMTU: ptr.To("-1500")}, field.NewPath("config"),
+			ConsistOf(PointTo(MatchFields(IgnoreExtras, Fields{"Field": Equal("config.vethMTU")})))),
+		Entry("should succeed with valid vethMTU", &apiscalico.NetworkConfig{VethMTU: ptr.To("1500")}, field.NewPath("config"),
+			BeEmpty()),
+		Entry("should return error with invalid vethMTU format", &apiscalico.NetworkConfig{VethMTU: ptr.To("invalid")}, field.NewPath("config"),
+			ConsistOf(PointTo(MatchFields(IgnoreExtras, Fields{"Field": Equal("config.vethMTU")})))),
+		Entry("should return error with invalid autoscaling mode", &apiscalico.NetworkConfig{
+			AutoScaling: &apiscalico.AutoScaling{
+				Mode: "invalid-mode",
+			},
+		}, field.NewPath("config"),
+			ConsistOf(PointTo(MatchFields(IgnoreExtras, Fields{"Field": Equal("config.autoScaling.mode")})))),
+
+		Entry("should succeed with valid autoscaling mode and resources", &apiscalico.NetworkConfig{
+			AutoScaling: &apiscalico.AutoScaling{
+				Mode: apiscalico.AutoscalingModeStatic,
+				Resources: &apiscalico.StaticResources{
+					Node: &corev1.ResourceList{
+						corev1.ResourceCPU:    resource.MustParse("100m"),
+						corev1.ResourceMemory: resource.MustParse("128Mi"),
+					},
+					Typha: &corev1.ResourceList{
+						corev1.ResourceCPU:    resource.MustParse("50m"),
+						corev1.ResourceMemory: resource.MustParse("64Mi"),
+					},
+				},
+			},
+		}, field.NewPath("config"),
+			BeEmpty()),
+		Entry("should fail with a static autoscaling mode and invalid resource name", &apiscalico.NetworkConfig{
+			AutoScaling: &apiscalico.AutoScaling{
+				Mode: apiscalico.AutoscalingModeStatic,
+				Resources: &apiscalico.StaticResources{
+					Node: &corev1.ResourceList{
+						"invalid": resource.MustParse("128Mi"),
+					},
+					Typha: &corev1.ResourceList{
+						corev1.ResourceCPU:    resource.MustParse("50m"),
+						corev1.ResourceMemory: resource.MustParse("64Mi"),
+					},
+				},
+			},
+		}, field.NewPath("config"),
+			ConsistOf(PointTo(MatchFields(IgnoreExtras, Fields{
+				"Field": Equal("config.autoScaling.resources.node.invalid"),
+			})))),
+		Entry("should return error with invalid IPIP mode", &apiscalico.NetworkConfig{IPIP: ptr.To(apiscalico.PoolMode("invalid"))}, field.NewPath("config"),
+			ConsistOf(PointTo(MatchFields(IgnoreExtras, Fields{"Field": Equal("config.ipip")})))),
+		Entry("should succeed with valid IPIP mode", &apiscalico.NetworkConfig{IPIP: ptr.To(apiscalico.Always)}, field.NewPath("config"),
+			BeEmpty()),
+		Entry("should succeed with valid IP autodetection method", &apiscalico.NetworkConfig{IPAutoDetectionMethod: ptr.To("interface=cali1234")}, field.NewPath("config"),
+			BeEmpty()),
```

Comment:
Should the `autoDetectionMethod` check not test the non-deprecated fields instead or do we want to test all of them?
WDYT?